### DISCUSSION TITLE
New beacons are not bringed to standard

### DISCUSSION THREAD

**Discussion by Xardion495** [FakeFishGames/Barotrauma#14029] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

On the new beacons, the properties of the installations like the junction boxes, nuclear reactors are configured incorrectly. Their deterioration speed should be reduced to 0.

### Reproduction steps

Check the properties in sub editor

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report. Why do you think the deterioration speed should be set to 0?

**Reply by Xardion495** [reply]
I'm going off the assumption that all buildings should follow a standard design. And the first beacon stations are like prefabs in that regard. If you look at Morko (formerly known as BeaconStation1), you'll see that the deterioration and consumption values are set to zero. That makes sense, because the beacons are mission elements that should only change based on player and environmental interactions, not on time.

Also, take a look at Ajokki - the junction boxes deterioration value is set to zero there too, but someone forgot to zero out the reactor value.

It looks like for the new beacon stations forgot to change the default settings when editing them.

**Reply by ubertpendragon** [reply]
Items in beacons are set to invulnerable after your repair them so it doesn't matter no?

I also have an opposite opinion:
All beacons are already wired and automated, unlike the ORIGINAL beacon (Morko) so there's really not much to do.

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/c973e36c-f439-4f50-84d5-fa8af4928d33)

This means less "repairing" of the beacon, since it pretty much drains itself automatically once power is on for example. 

Same for transducers not being used by sonar monitors (which you actually can't interact with because of the small piece of wall that was added on the antenna), you don't have to rewire them if they get disconnected

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/84757a6f-bd58-4447-b0f4-d2a622150fd0)



A small change would be to add the water detector and sigchecks to the Bandit layer: you'd have to manually power the pumps on broken beacons, but they would drain automatically if bandits are present

**Reply by Xardion495** [reply]
> Items in beacons are set to invulnerable after your repair them so it doesn't matter no?

From a game editor point, that's not obvious. And of course, adding special methods to deal with such cases instead of just set initially suitable values leads to some pretty dark thoughts about game code optimization.
Do you always stick to such ways of solving similar situations?
Have you heard that it's always best to keep the number of function calls down to a minimum?

> All beacons are already wired and automated, unlike the ORIGINAL beacon (Morko) so there's really not much to do.
> This means less "repairing" of the beacon, since it pretty much drains itself automatically once power is on for example.

Not sure what you're trying to say. All beacons have marked checkbox "Allow disconnected wires". Wired/automated beacons or not, players have to (forced to) maintaining them and this is a pretty intense game situation. For most player wiring the beacon items is not trivial.
![image](https://github.com/FakeFishGames/Barotrauma/assets/17722887/44029373-1859-4d93-b38a-89dc7c8876a0)

> Same for transducers not being used by sonar monitors (which you actually can't interact with because of the small piece of wall that was added on the antenna), you don't have to rewire them if they get disconnected

Sounds weird. Starting from the second biome, I periodically find the transducers disconnected from the sonar monitor and from the power grid and have to maintaining them. Are we play the same game?

Anyway. All building items must be configured to a single design. Setting the values for deterioration and consumption to zero means that the beacons will only be affected by player actions and it will have less impact on performance because its state will be more static.

**Reply by Regalis11** [reply]
While I agree that from a design perspective things should be kept consistent, in this case I feel like this is mostly a theoretical issue. The players don't generally spend nowhere near enough time repairing the beacon station for the deterioration to even matter.

As for optimization, I don't think that's a valid argument. The performance cost of the deterioration is completely negligible, so there's absolutely no need to disable it for performance reasons. Disabling it also does not affect the number of function calls as you suggest.

**Reply by Xardion495** [reply]
So I have nothing else to say. Closing the ticket
